Unity-Technologies/createstudio_jenkins_lib
# Prevent long commit messages from truncating build reports on Slack

### Purpose of this PR

This PR checks the length of last commit message and adjusts the Slack build report accordingly.

1. If the message is less than `max_commit_length` it will show the full commit message
2. if the message exceeds the limit:
   * It will display a preview in the main report
   * It will attach the full commit message in a thread

### Testing status

Tested by creating a branch of `createstudio_core`. Set the to `SLACK_CHANNEL` to `createstudio-builds` and tested multiple scenarios:

- Build with commit message that is under the limit (successful + failed)
- Build with commit message that exceeds the limit (successful + failed)

It appears to be working as expected.

### Todo

How long should we allow the commit message to be? Its currently set to 128 but we could make it a few hundred chars if we want.
